src/Jasper.Diagnostics/DiagnosticsBusLogger.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using Jasper.Diagnostics.Messages;
+using Jasper.Remotes.Messaging;
+using JasperBus;
+using JasperBus.Runtime;
+
+namespace Jasper.Diagnostics
+{
+    public class DiagnosticsBusLogger : IBusLogger
+    {
+        private readonly IDiagnosticsClient _client;
+
+        public DiagnosticsBusLogger(IDiagnosticsClient client)
+        {
+            _client = client;
+        }
+
+        public void ExecutionFinished(Envelope envelope)
+        {
+        }
+
+        public void ExecutionStarted(Envelope envelope)
+        {
+        }
+
+        public void LogException(Exception ex, string correlationId = null, string message = "Exception detected:")
+        {
+        }
+
+        public void MessageFailed(Envelope envelope, Exception ex)
+        {
+            _client.Send(new MessageFailed(envelope, ex));
+        }
+
+        public void MessageSucceeded(Envelope envelope)
+        {
+            _client.Send(new MessageSucceeded(envelope));
+        }
+
+        public void NoHandlerFor(Envelope envelope)
+        {
+        }
+
+        public void Received(Envelope envelope)
+        {
+        }
+
+        public void Sent(Envelope envelope)
```

Comment:
I'd never thought of this before, but it might be cool to track what messages are sent out too.